bin/download_release_dependencies.py

```
@@ -0,0 +1,164 @@
+        # On Linux, GHA runner can have 7z installed, or we use our downloaded 7zz
+        seven_zip_path = base_dir / "bin" / "7z" / "linux" / "amd64" / "7zz"
+        # If it doesn't exist, try standard system 7z
+        if not seven_zip_path.exists():
+            seven_zip_path = Path(shutil.which("7z") or "7z")
+
+    if not seven_zip_path.exists() and not shutil.which("7z"):
+        raise FileNotFoundError("7-Zip binary not found. Please run download_all_build_binaries.py first.")
+
+    import subprocess
+    cmd = [str(seven_zip_path), "x", str(archive_path), f"-o{extract_to}", "-y"]
+    subprocess.run(cmd, check=True)
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)